### ISSUE TITLE
BitGOJS Browser Library - Unable to compile a single file to use in the browser

### ISSUE THREAD

**Issue by MrPlumbum82** [BitGo/BitGoJS#42]
I'm having a hard time trying to get my hands on the BitGo Client-side JavaScript Library (for Browsers). I'll explain below the steps I took and the outcome of those steps.

1. Version Check
![1_versions](https://cloud.githubusercontent.com/assets/11234396/25609288/02e0f022-2f38-11e7-9a05-d0ac398c6f78.png)

2. Cloning the Repo
![2_clone](https://cloud.githubusercontent.com/assets/11234396/25609314/1a9e4d0e-2f38-11e7-8e7c-eb764c4d0475.png)

3. ` npm install`
Below are sections where I saw errors or suspicious output
![3_npm_install](https://cloud.githubusercontent.com/assets/11234396/25609348/4a5c424e-2f38-11e7-8a0f-10c6f9ab7f67.png)
![4_unsupported_platform](https://cloud.githubusercontent.com/assets/11234396/25609353/51ef290e-2f38-11e7-9bc2-74dd904d88f4.png)

4. ` npm run-script compile`
![5_compile](https://cloud.githubusercontent.com/assets/11234396/25609380/737cfaba-2f38-11e7-9a8c-4bc2fca12893.png)

5. ` npm run-script compile-dbg`
![6_compile_dbg](https://cloud.githubusercontent.com/assets/11234396/25609390/800fcb18-2f38-11e7-8dde-764d26fd83d1.png)

What am I doing wrong? My goal is to compile a single JavaScript library file to use in the browser. Am I on the right track? 

npm-debug.log file
https://ufile.io/9cksn

**Comment by MrPlumbum82** [comment]
Update (tried again with a different nodeJS Version, details in the screens)

1. Version Check
![1_versions](https://cloud.githubusercontent.com/assets/11234396/25693062/6a7954fc-30c0-11e7-8ec8-7a3f78d73ce6.png)


2. Cloning the Repo
![2_clone](https://cloud.githubusercontent.com/assets/11234396/25693063/6fc4d4e0-30c0-11e7-87ed-63a155eca784.png)


3. ` npm install`
![3_npm_install](https://cloud.githubusercontent.com/assets/11234396/25693070/7693c326-30c0-11e7-9eb0-e999a165e18a.png)


4. ` npm run-script compile`
![4_compile](https://cloud.githubusercontent.com/assets/11234396/25693135/ee74a478-30c0-11e7-91a5-0027be213871.png)



5. ` npm run-script compile-dbg`
![5_compile_dbg](https://cloud.githubusercontent.com/assets/11234396/25693137/f4d8a7c4-30c0-11e7-94ea-cb8930189ac1.png)


Any ideas?

npm-debug.log file
https://ufile.io/urkuc

**Comment by kuzcoh** [comment]
Hello,

We were unable to reproduce this behavior using the exact same versions of nvm,node, and npm. We believe the errors you are seeing are due to the 'npm install' command not exiting properly. Does  this behavior persist across multiple devices? Perhaps the node_modules file was not removed prior to running 'npm install'?

**Comment by MrPlumbum82** [comment]
**Attempt 2 info:**
As shown in the images above, the second time around I recloned the entire repo and ran npm install again. So there's no way the node_modules were not removed. Also if this helps here is the system info of the PC I used while attempting the compile above.
![first_device](https://cloud.githubusercontent.com/assets/11234396/26142168/d04f226a-3af8-11e7-8e7e-b37e25b51cc9.png)

**Attempt 3:**
Here's me trying to do the same on a Mac
![mac_info](https://cloud.githubusercontent.com/assets/11234396/26142138/a20198ca-3af8-11e7-9218-e525ba90c6d6.png)

![mac_1](https://cloud.githubusercontent.com/assets/11234396/26142136/a1faf9e8-3af8-11e7-8655-fb344d2c8caa.png)

Cloning and running npm install
![mac_2](https://cloud.githubusercontent.com/assets/11234396/26142137/a2002f94-3af8-11e7-9364-0a8ea5dd8943.png)

Running npm run-script compile
![mac_3](https://cloud.githubusercontent.com/assets/11234396/26142135/a1fa2522-3af8-11e7-896f-145eee8dccc5.png)


**Attempt 4:**
Here's me trying to do the same on a Windows 7 PC
![win_info](https://cloud.githubusercontent.com/assets/11234396/26142141/a223a1f4-3af8-11e7-97f5-0c4959ad2168.png)

After running npm install on the freshly cloned repo
![win_1](https://cloud.githubusercontent.com/assets/11234396/26142139/a2043436-3af8-11e7-9dd9-110565c9fc17.png)

Ran the npm run-script compile
![win_2](https://cloud.githubusercontent.com/assets/11234396/26142140/a207554e-3af8-11e7-96d4-b0f8d0414778.png)

Whats going on here? You guys aren't having this issue? How is that even possible? What am I doing wrong? Are their any debug files I could send you guys that would help you understand why this is happening?

Really all I need the client slide JavaScript library. Don't you guys host the compiled library somewhere on the website? Or could someone upload a compiled library file so I can start using the library?

**Comment by omaidf** [comment]
Hello,

We are now able to reproduce this behavior, could you please let us know if the following commands resolve your issue?

1. npm install bufferutil
2. npm install utf-8-validate
3. npm run-script compile-dbg
4. ls -l browser/

You should now see a full BitGoJS.js file!

**Comment by MrPlumbum82** [comment]
Thanks, that did it for me.

Just to clarifiy on omaidf's response for any people reading this issue in the future and still having the same issue I did, here is a step by step of what I did;

Version Numbers:
node v6.9.5
npm v3.10.10

(I tried this with node version 7.7.0 and it did not work so use the version mentioned above)

1. git clone https://github.com/BitGo/BitGoJS.git
2. cd BitGoJs
3. npm install
4. npm install bufferutil
5. npm install utf-8-validate
6. npm run-script compile-dbg

The complete library file was placed at browser/BitGoJS.js
